lib/cloud/awsconfig/awsconfig.go

```diff
@@ -302,6 +327,17 @@ func getAssumeRoleProvider(ctx context.Context, clt stscreds.AssumeRoleAPIClient
 		if role.ExternalID != "" {
 			aro.ExternalID = aws.String(role.ExternalID)
 		}
+		aro.RoleSessionName = maybeHashRoleSessionName(role.SessionName)
+		if role.Duration >= 15*time.Minute {
+			// minimum duration is 15 minutes.
+			aro.Duration = role.Duration
+		}
```

Comment:
what's the behavior when this expires? will `aws.NewCredentialsCache` just fetch again? as discussed offline, we don't have to set the duration.